lib/report_generator.rb

```diff
@@ -40,8 +40,20 @@ def self.params_to_hash(query_params)
       params_hash
     end
 
-    def self.build_report_pms(query, table = "", targets = [])
+    def self.build_report_pms(query, table = "", targets = [], opts: {})
```

Comment:
I think since the options are few and well known (i.e. not forwarded from some other interface) we could spell them out as keyword arguments.

```suggestion
    def self.build_report_pms(query, table = "", targets = [], result:, attach_csv:)
```

Also I think if we make the `result` object a bit richer, it could provide the `query` (and perhaps the `table` as well.)

WDYT?